sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -113,11 +113,11 @@ case class CreateViewCommand(
     verifyTemporaryObjectsNotExists(catalog, isTemporary, name, child)
 
     if (viewType == LocalTempView) {
-      val samePlan = catalog.getTempView(name.table).exists {
+      val samePlan = catalog.getTempView(name.table).map {
         // Don't perform sameResult check for View logical plan, since it's unresolved
         case _: View => false
         case other => other.sameResult(child)
-      }
+      }.getOrElse(true)
       if (replace && !samePlan) {
```

Comment:
Looks good but there's a nit here. The name `samePlan` sounds not correct now  because it checks if the view exists or not. What about something like this?

```scala
...
val shouldUncache = replace && catalog.getGlobalTempView(name.table).exists {
  // Uncache View logical plan, since it's unresolved.
  case _: View => true
  case other => !other.sameResult(child)
}
if (shouldUncache) {
...
```

Reply:
This seems better. Thanks @HyukjinKwon!